packages/rsocket-examples/src/ClientServerRequestChannelResumeExample.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  Cancellable,
+  OnExtensionSubscriber,
+  OnNextSubscriber,
+  OnTerminalSubscriber,
+  Payload,
+  Requestable,
+  RSocket,
+  RSocketConnector,
+  RSocketServer,
+} from "@rsocket/rsocket-core";
+import { exit } from "process";
+import { TcpClientTransport } from "@rsocket/rsocket-tcp-client";
+import { TcpServerTransport } from "@rsocket/rsocket-tcp-server";
+
+async function main() {
+  const server = new RSocketServer({
+    transport: new TcpServerTransport({
+      listenOptions: {
+        port: 8000,
+      },
+    }),
+    acceptor: {
+      accept: async (): Promise<Partial<RSocket>> => {
+        return {
+          requestChannel: (
+            payload: Payload,
+            initialRequestN: number,
+            isCompleted: boolean,
+            responderStream: OnTerminalSubscriber &
+              OnNextSubscriber &
+              OnExtensionSubscriber &
+              Requestable &
+              Cancellable
+          ): OnTerminalSubscriber &
+            OnNextSubscriber &
+            OnExtensionSubscriber &
+            Requestable &
+            Cancellable => {
+            responderStream.onNext(payload, isCompleted);
+            responderStream.request(initialRequestN);
+
+            return {
+              cancel(): void {
+                responderStream.cancel();
+              },
+              onComplete(): void {
+                responderStream.onComplete();
+              },
+              onError(error: Error): void {
+                responderStream.onError(error);
+              },
+              onExtension(): void {},
+              onNext(payload: Payload, isComplete: boolean): void {
+                setTimeout(
+                  () => responderStream.onNext(payload, isComplete),
+                  10
+                );
+              },
+              request(requestN: number): void {
+                setTimeout(() => responderStream.request(requestN), 1);
+              },
+            };
+          },
+        };
+      },
+    },
+    resume: {
+      sessionTimeout: 60 * 1000, // 60sec
+    },
+  });
+  const connector = new RSocketConnector({
+    setup: {
+      keepAlive: 100,
+      lifetime: 10000,
+    },
+    resume: {
+      tokenGenerator: () => Buffer.from("1"),
+      reconnectFunction: (a) =>
+        new Promise((r) => setTimeout(r, a * 100 + 100, 100)),
+    },
```

Comment:
Is this example intended to demonstrate the `Resume` functionality, or more simply just the connector `Resume` config?

Reply:
This example demonstrates the complete connector-server resume example, but it has to be supported with better docs. E.g. to have it running one has to run socat on 8000 <-> 8001 ports
and then terminating + reestablishing socat connection we can see resume working